File: ruby/grains/grains.rb
# A grain to feed us all
class Grains
  class << self
    def square(number)
      raise ArgumentError unless number.between?(1, 64)
      return 1 if number.eql? 1

      2 * square(number - 1)
    end

    def total
      (1..64).reduce(0) do |sum, square|
        sum + square(square)
      end
    end
  end
end

File: go/grains/grains.go
// Package grains implements the grains exercise from exercism.io
package grains

import "errors"

// Total calculate total number of grains on the chessboard
func Total() uint64 {
	total := uint64(1)
	for i := 1; i < 64; i++ {
		total += 1 << uint64(i)
	}

	return total
}

// Square calculates the total number of square on a chessboard square
func Square(position int) (uint64, error) {
	if position < 1 || position > 64 {
		return 0, errors.New("Square position invalid")
	}

	return 1 << uint64(position-1), nil
}
